libs/react-spa/bff/src/lib/BffProvider.tsx

```diff
@@ -0,0 +1,240 @@
+      })
+    }
+  }, [postMessage, state.userInfo, isLoggedIn])
+
+  const checkLogin = async (noRefresh = false) => {
+    dispatch({
+      type: ActionType.SIGNIN_START,
+    })
+
+    try {
+      const url = bffUrlGenerator('/user', {
+        no_refresh: noRefresh.toString(),
+      })
+
+      const res = await fetch(url, {
+        credentials: 'include',
+      })
+
+      if (!res.ok) {
+        // Bff server is down
+        if (res.status >= 500) {
+          throw new Error(BFF_SERVER_UNAVAILABLE)
+        }
+
+        // For other none ok responses, like 401/403, proceed with sign-in redirect.
+        signIn()
+
+        return
+      }
+
+      const user = await res.json()
+
+      dispatch({
+        type: ActionType.SIGNIN_SUCCESS,
+        payload: user,
+      })
+    } catch (error) {
+      dispatch({
+        type: ActionType.ERROR,
+        payload: error,
+      })
+    }
+  }
+
+  const signIn = useCallback(() => {
+    dispatch({
+      type: ActionType.SIGNIN_START,
+    })
+
+    window.location.href = bffUrlGenerator('/login', {
+      target_link_uri: window.location.href,
+    })
+  }, [bffUrlGenerator])
+
+  const signOut = useCallback(() => {
+    if (!state.userInfo) {
+      return
+    }
+
+    dispatch({
+      type: ActionType.LOGGING_OUT,
+    })
+
+    window.location.href = bffUrlGenerator('/logout', {
+      sid: state.userInfo.profile.sid,
+    })
+
+    setTimeout(() => {
+      // We will wait 5 seconds before we post the logout message to other tabs/windows/iframes.
+      // The reason is that IDS will not log the user out immediately.
+      postMessage({
+        type: BffBroadcastEvents.LOGOUT,
+      })
+    }, 5000)
+  }, [bffUrlGenerator, postMessage, state.userInfo])
+
+  const switchUser = (nationalId?: string) => {
+    dispatch({
+      type: ActionType.SWITCH_USER,
+    })
+
+    window.location.href = bffUrlGenerator(
+      '/login',
+      nationalId
+        ? {
+            login_hint: nationalId,
+          }
+        : {
+            prompt: 'select_account',
+          },
+    )
+  }
+
+  const checkQueryStringError = () => {
+    const urlParams = new URLSearchParams(window.location.search)
+    const error = urlParams.get('bff_error_code')
+    const errorDescription = urlParams.get('bff_error_description')
+
+    if (error) {
+      dispatch({
+        type: ActionType.ERROR,
+        payload: new Error(`${error}: ${errorDescription}`),
+      })
+    }
+
+    // Returns true if there is an error
+    return !!error
+  }
+
+  useEffectOnce(() => {
+    const hasError = checkQueryStringError()
+
+    if (!hasError) {
+      checkLogin()
+    }
+  })
+
+  useEffectOnce(() => {
+    const timeout = setTimeout(() => {
+      // After one hour we check if the user is still logged in
+      // and we tell the /user endpoint not to refresh the tokens,
+      // since we are checking for timeout expiration.
+      checkLogin(true)
+    }, ONE_HOUR_MS)
```

Comment:
This is kind of redundant with the BffPoller. Let's get rid of it for now. I also don't like hard-coding one-hour in too many places. Ideally it is set in an environment variable and propagated to the client through the /bff/user endpoint **if** needed.